lib/ctcs2_to_junit.pm

```diff
@@ -0,0 +1,118 @@
+# SUSE's openQA tests
+#
+# Copyright © 2017 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary:  base class for convert QA:Head ctcs2 wrapper testsuite test log to junit format
+# Maintainer: Yong Sun <yosun@suse.com>
+
+package ctcs2_to_junit;
+
+use strict;
+use warnings;
+use base "Exporter";
+use Exporter;
+use testapi;
+use utils;
+use XML::Writer;
+
+our @EXPORT = qw(analyzeResult generateXML);
+
+sub analyzeResult {
+    my ($text) = @_;
+    my $result;
+    $text =~ /Test in progress(.*)Test run complete/s;
+    my $rough_result = $1;
+    foreach (split("\n", $rough_result)) {
+        if ($_ =~ /(\S+)\s+\.{3}\s+\.{3}\s+(PASSED|FAILED|SKIPPED)\s+\((\S+)\)/g) {
+            $result->{$1}{status} = $2;
+            $result->{$1}{time}   = $3;
+        }
+    }
+    return $result;
+}
+
+sub generateXML {
+    my ($data) = @_;
+
+    my %my_hash = %$data;
+    my $case_status;
+    my $case_num = scalar(keys %my_hash);
+    my $pass_num = 0;
+    my $fail_num = 0;
+    my $skip_num = 0;
+    my $writer   = new XML::Writer(DATA_MODE => 'true', DATA_INDENT => 2, OUTPUT => "self");
+
+    foreach my $item (keys(%my_hash)) {
+        if ($my_hash{$item}->{status} =~ m/PASSED/) {
+            $pass_num += 1;
+        }
+        elsif ($my_hash{$item}->{status} =~ m/SKIPPED/) {
+            $skip_num += 1;
+        }
+        else {
+            $fail_num += 1;
+        }
+    }
+    #my $count = $pass_nums + $fail_nums + $skip_nums;
```

Comment:
Please avoid code in comment, if it's not necessary why do you need to keep it? Could be useful during development but not in a PR.
At least, add a comment to explain why you need to keep it.

Reply:
ah, forgot to remove it.